File: ePozoriste.Model/Predstava.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ePozoriste.Model
{
    public class Predstava
    {
        public int PredstavaId { get; set; }
        public string Naziv { get; set; }
        public string Sadrzaj { get; set; }
        public string? Slika { get; set; }
        public int VrijemeTrajanje { get; set; }
        public string Rezija { get; set; }
        public string Scenografija { get; set; }
        public string Kostimografija { get; set; }

        public override string ToString()
        {
            return Naziv;
        }
    }
}

File: epozoriste_mobile/lib/models/predstava.dart
import 'package:epozoriste_mobile/models/models.dart';
import 'package:json_annotation/json_annotation.dart';

part 'predstava.g.dart';

@JsonSerializable()
class Predstava {
  int predstavaId;
  String naziv;
  String sadrzaj;
  String? slika;
  int vrijemeTrajanje;
  String rezija;
  String scenografija;
  String kostimografija;
  int? vrstaPredstaveId;
  VrstaPredstave? vrstaPredstave;

  Predstava(
      {required this.predstavaId,
      required this.naziv,
      required this.sadrzaj,
      this.slika,
      required this.vrijemeTrajanje,
      required this.rezija,
      required this.scenografija,
      required this.kostimografija,
      this.vrstaPredstaveId,
      this.vrstaPredstave});

  factory Predstava.fromJson(Map<String, dynamic> json) =>
      _$PredstavaFromJson(json);
  Map<String, dynamic> toJson() => _$PredstavaToJson(this);
}
